tests/benchmark/test_worst_blocks.py

```diff
@@ -330,3 +332,50 @@ def test_block_full_access_list_and_data(
             access_list=access_list,
         ),
     )
+
+
+@pytest.mark.parametrize("empty_authority", [True, False])
+@pytest.mark.parametrize("zero_delegation", [True, False])
+def test_worst_case_auth_block(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    intrinsic_cost: int,
+    gas_benchmark_value: int,
+    fork: Fork,
+    empty_authority: bool,
+    zero_delegation: bool,
+):
+    """Test an auth block."""
+    gas_costs = fork.gas_costs()
+
+    iteration_count = (gas_benchmark_value - intrinsic_cost) // gas_costs.G_AUTHORIZATION
+
+    code = Op.STOP + Op.JUMPDEST * (fork.max_code_size() - 1)
+    auth_target = pre.deploy_contract(code=code)
+
+    auth_tuples = [
+        AuthorizationTuple(
+            address=Address(0) if zero_delegation else auth_target,
+            nonce=0,
```

Comment:
I think the nonce is a bit tricky here, because if you check the logic in [here](https://github.com/ethereum/execution-spec-tests/blob/ef11a98dd378e9ec07f86edc31246e826b9efff8/src/pytest_plugins/filler/pre_alloc.py#L223-L241) it's not super simple.
My suggestion would be to rework this as a loop, similar to:
```python
for _ in range(iteration_count):
    signer = pre.fund_eoa(delegation=None) if empty_authority else pre.fund_eoa(delegation=Address(1))
    auth_tuple = AuthorizationTuple(..., nonce=signer.nonce, ...)
    auth_tuples.append(auth_tuple)
```